### PR TITLE
feat: disable 'Change PIN' button if a PIN input is empty

### PR DESCRIPTION
**Pull request by SophiaSales**
# Summary of Changes
Captura de tela de 2023-09-12 14-46-46 Captura de tela de 2023-09-12 14-45-47

If any of the PIN inputs (current, new and confirm new) are empty, the 'Change PIN' button should be disabled.

# Related Issues

N/A.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [X] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [X] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [X] Updated documentation as needed for changed code and new or modified features;
- [X] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: packages/legacy/core/App/screens/PINCreate.tsx [hunk 1]
```diff
@@ -163,6 +164,10 @@ const PINCreate: React.FC<PINCreateProps> = ({ setAuthenticated, route }) => {
     await passcodeCreate(PINOne)
   }
 
+  const checkEmptyPINs = (p: string) => {
+    return p !== '' && PIN !== '' && PINTwo !== ''
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/PINCreate.tsx]: Just noticing there is no check for PINOld here

File: packages/legacy/core/App/screens/PINCreate.tsx [hunk 2]
```diff
@@ -180,13 +185,15 @@ const PINCreate: React.FC<PINCreateProps> = ({ setAuthenticated, route }) => {
...
                 setPINOld(p)
+                setContinueEnabled(checkEmptyPINs(p))
               }}
             />
           )}
           <PINInput
             label={t('PINCreate.EnterPINTitle', { new: updatePin ? t('PINCreate.NewPIN') + ' ' : '' })}
             onPINChanged={(p: string) => {
               setPIN(p)
+              setContinueEnabled(checkEmptyPINs(p))
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/PINCreate.tsx]: Should this `p` be disregarded and just include a check in `checkEmptyPINs` for `PINOld`?

File: packages/legacy/core/App/screens/PINCreate.tsx [hunk 3]
```diff
@@ -227,6 +234,7 @@ const PINCreate: React.FC<PINCreateProps> = ({ setAuthenticated, route }) => {
             label={t('PINCreate.ReenterPIN', { new: updatePin ? t('PINCreate.NewPIN') + ' ' : '' })}
             onPINChanged={(p: string) => {
               setPINTwo(p)
+              setContinueEnabled(checkEmptyPINs(p))
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/PINCreate.tsx]: Same as above

File: packages/legacy/core/App/screens/PINCreate.tsx [hunk 4]
```diff
@@ -180,13 +185,15 @@ const PINCreate: React.FC<PINCreateProps> = ({ setAuthenticated, route }) => {
               testID={testIdWithKey('EnterOldPIN')}
               onPINChanged={(p: string) => {
                 setPINOld(p)
+                setContinueEnabled(checkEmptyPINs(p))
```
Comment (bryce-mcmath) [inline packages/legacy/core/App/screens/PINCreate.tsx]: Same as below

**Review by bryce-mcmath — CHANGES_REQUESTED** [review]

**Comment by thiagoromanos** [comment]
Suggestion for this PR: you could use a `useEffect` function watching `[PINOld, PIN, PINTwo]` that validates those values if `updatePin == true`, instead of watching every input change.
What do you think, @SophiaSales and @bryce-mcmath ?

**Comment by bryce-mcmath** [comment]
> Suggestion for this PR: you could use a `useEffect` function watching `[PINOld, PIN, PINTwo]` that validates those values if `updatePin == true`, instead of watching every input change. What do you think, @SophiaSales and @bryce-mcmath ?

That works for me!

**Review by bryce-mcmath — APPROVED** [review]

**Comment by SophiaSales** [comment]
@bryce-mcmath, I have no write permission. Can you merge this PR for me?

**Comment by bryce-mcmath** [comment]
> @bryce-mcmath, I have no write permission. Can you merge this PR for me?

Done! Nice work.

**Pull request merged**